File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Python/s936984238_comments_removed.py
import sys

def main():
    sc = Scanner()
    
    while read(sc):
        solve()

def read(sc):
    number = sc.nextInt()
    if number == 0:
        return False
    
    score = [sc.nextInt() for _ in range(number)]
    global scores
    scores = score
    return True

def solve():
    scores.sort()
    sum = 0
    for i in range(1, len(scores) - 1):
        sum += scores[i]
    print(sum // (len(scores) - 2))

class Scanner:
    def __init__(self):
        self.tokens = []
        self.token_index = 0
        self.input = sys.stdin

    def nextInt(self):
        if self.token_index == len(self.tokens):
            self.tokens = self.input.readline().strip().split()
            self.token_index = 0
        token = self.tokens[self.token_index]
        self.token_index += 1
        return int(token)

if __name__ == "__main__":
    main()  # End of Code

File: tasks/translation/filtered_dataset/codenet/Java/Code/s936984238_comments_removed.java
import java.util.*;
public class Main {
	static Scanner sc = new Scanner(System.in);

	static int[] score;
	static int number;
	public static void main(String[] args) {


		while(read()){
			slove();
		}
	}
	static boolean read(){
		number = sc.nextInt();
		if(number == 0)
			return false;
		
		score = new int[number];
		for(int i = 0; i < number; i++){
			score[i] = sc.nextInt();
		}
		return true;
	}
	static void slove(){
		java.util.Arrays.sort(score);
		int sum = 0;
		for(int i = 1; i < number-1; i++){
			sum = sum + score[i];
		}
		System.out.println(sum/(number-2));
	}
}